stores/src/main/java/com/issoft/training/stores/comparaton/ProductComparator.java

```diff
@@ -3,41 +3,52 @@
 import com.issoft.training.domain.shop.Product;
 
 import java.util.Comparator;
+import java.util.List;
 
 public class ProductComparator implements Comparator<Product> {
+    private final List<String> orderingValue;
+    private final List<String> orderingName;
 
-    private final String orderingName;
-    private final String orderingPrice;
-    private final String orderingRate;
-
-    public ProductComparator(String orderingName, String orderingPrice, String orderingRate) {
+    public ProductComparator(List<String> orderingValue, List<String> orderingName) {
+        this.orderingValue = orderingValue;
         this.orderingName = orderingName;
-        this.orderingPrice = orderingPrice;
-        this.orderingRate = orderingRate;
     }
 
     public int compare(Product product1, Product product2) {
         int i;
-        if (orderingName.equals("asc")) {
-            i = product1.getName().compareToIgnoreCase(product2.getName());
-        } else {
-            i = product2.getName().compareToIgnoreCase(product1.getName());
-        }
-        if (i != 0) {
-            return i;
-        }
-        if (orderingPrice.equals("asc")) {
-            i = Double.compare(product1.getPrice(), product2.getPrice());
-        } else {
-            i = Double.compare(product2.getPrice(), product1.getPrice());
-        }
-        if (i != 0) {
-            return i;
-        }
-        if (orderingRate.equals("asc")) {
-            i = Double.compare(product1.getRate(), product2.getRate());
-        } else {
-            i = Double.compare(product2.getRate(), product1.getRate());
+        for (String name : orderingName) {
```

Comment:
very strange code, why do you iterate over second array x times... nevermind, when you move to map impl there will be only one for